app/models/course/survey/answer.rb

```diff
@@ -3,7 +3,20 @@ class Course::Survey::Answer < ActiveRecord::Base
   belongs_to :response, inverse_of: :answers
   belongs_to :question, inverse_of: :answers
   has_many :options, class_name: Course::Survey::AnswerOption.name,
-                     inverse_of: :answer, dependent: :destroy
+                     inverse_of: :answer, dependent: :destroy, autosave: true
 
-  accepts_nested_attributes_for :options
+  def selected_options=(question_option_ids)
```

Comment:
I'm not sure if we need to be so complex here, I think we can just set the params to `option_ids`, then assign the `ids` as part of an array. Did you consider that?